### PR TITLE
Fixed Triggering onSelect twice on Dropdown Menu

### PR DESCRIPTION
**Pull request by hassanzohdy**
The issue is `onSelect` is triggered twice.

```tsx
export default function Dropdown() {
  const handleSelect = (eventKey: any) => {
    console.log(eventKey);
  };
  return (
    <>
      <Dropdown.Menu onSelect={handleSelect}>
        <Dropdown.Item eventKey={3}>Download As...</Dropdown.Item>
        <Dropdown.Item eventKey={4}>Export PDF</Dropdown.Item>
        <Dropdown.Item eventKey={5}>Export HTML</Dropdown.Item>
        <Dropdown.Item eventKey={6}>Settings</Dropdown.Item>
        <Dropdown.Item eventKey={7}>About</Dropdown.Item>
      </Dropdown.Menu>
    </>
  );
}
```

Or Check the following Code Sandbox 

https://codesandbox.io/s/react-typescript-forked-iybjwt?file=/src/App.tsx

Click on any item on the dropdown then check the console log

### CODE REVIEW

File: src/Dropdown/test/DropdownMenuSpec.js [hunk 1]
```diff
@@ -234,6 +234,30 @@ describe('<Dropdown.Menu>', () => {
...
+      <DropdownMenu
+        onSelect={eventKey => {
+          selectedValues.push(eventKey);
+        }}
+      >
+        <DropdownItem eventKey={1}>1</DropdownItem>
+        <DropdownItem eventKey={2}>2</DropdownItem>
+        <DropdownItem eventKey={3}>3</DropdownItem>
+      </DropdownMenu>
+    );
```
Comment (SevenOutman) [inline src/Dropdown/test/DropdownMenuSpec.js:250]: Please use `@testing-library/react` to render components in new test cases. Example here https://github.com/rsuite/rsuite/blob/d5b67383b5/src/Dropdown/test/DropdownMenuItemSpec.js#L12-L16

File: src/Dropdown/test/DropdownMenuSpec.js [hunk 2]
```diff
@@ -234,6 +234,30 @@ describe('<Dropdown.Menu>', () => {
...
+
+    act(() => {
+      Simulate.click(instance.querySelectorAll('[role^="menuitem"]')[2], {
+        bubbles: true
+      });
+    });
+
+    expect(selectedValues.length).to.be.eq(1);
```
Comment (SevenOutman) [inline src/Dropdown/test/DropdownMenuSpec.js:258]: Please use `sinon-chai` assertions to assert function called and arguments. Example here https://github.com/rsuite/rsuite/blob/d5b67383b5/src/Dropdown/test/DropdownMenuItemSpec.js#L101

File: src/Dropdown/test/DropdownMenuSpec.js [hunk 3]
```diff
@@ -234,6 +234,30 @@ describe('<Dropdown.Menu>', () => {
...
+        <DropdownItem eventKey={2}>2</DropdownItem>
+        <DropdownItem eventKey={3}>3</DropdownItem>
+      </DropdownMenu>
+    );
+
+    act(() => {
+      Simulate.click(instance.querySelectorAll('[role^="menuitem"]')[2], {
+        bubbles: true
+      });
+    });
```
Comment (SevenOutman) [inline src/Dropdown/test/DropdownMenuSpec.js:256]: Please use `@testing-library/user-event` to interact with elements in new test cases. Example here https://github.com/rsuite/rsuite/blob/d5b67383b5/src/Dropdown/test/DropdownMenuItemSpec.js#L99

**Review by SevenOutman — CHANGES_REQUESTED** [review]

File: src/Dropdown/test/DropdownMenuSpec.js (hunk 2 above)
Comment (SevenOutman) [reply src/Dropdown/test/DropdownMenuSpec.js:262]: > If there is any other way to use `sinon-chai` to handle that case please guide me to it.

There's a [property `.callCount`](https://sinonjs.org/releases/latest/spies/) recording how many times a spy functions has been called.

File: src/Dropdown/test/DropdownMenuSpec.js [hunk 4]
```diff
@@ -234,6 +234,30 @@ describe('<Dropdown.Menu>', () => {
...
+      <DropdownMenu onSelect={onSelectSpy}>
+        <DropdownItem data-testid="item-1" eventKey={1}>
+          1
+        </DropdownItem>
+        <DropdownItem data-testid="item-2" eventKey={2}>
+          2
+        </DropdownItem>
+        <DropdownItem data-testid="item-3" eventKey={3}>
+          3
+        </DropdownItem>
```
Comment (SevenOutman) [inline src/Dropdown/test/DropdownMenuSpec.js:250]: No need to specify `data-testid` if these elements are not going to be queried.

File: src/Dropdown/test/DropdownMenuSpec.js [hunk 5]
```diff
@@ -234,6 +234,30 @@ describe('<Dropdown.Menu>', () => {
...
+      </DropdownMenu>
+    );
+
+    act(() => {
+      userEvent.click(getByTestId('item-1'));
+    });
```
Comment (SevenOutman) [inline src/Dropdown/test/DropdownMenuSpec.js:256]: `userEvent` methods does `act()` automatically. You can omit this explicit `act()` call

File: src/Dropdown/test/DropdownMenuSpec.js [hunk 6]
```diff
@@ -234,6 +234,30 @@ describe('<Dropdown.Menu>', () => {
...
+
+    expect(onSelectSpy.callCount).to.be.eq(1);
```
Comment (SevenOutman) [inline src/Dropdown/test/DropdownMenuSpec.js:258]: The test case title mentioned `eventKey` correctness but it's not asserted here.

**Review by SevenOutman — CHANGES_REQUESTED** [review]

File: src/Dropdown/test/DropdownMenuSpec.js (hunk 6 above)
Comment (hassanzohdy) [reply src/Dropdown/test/DropdownMenuSpec.js:252]: Done

File: src/Dropdown/test/DropdownMenuSpec.js (hunk 5 above)
Comment (hassanzohdy) [reply src/Dropdown/test/DropdownMenuSpec.js]: Done

**Review by SevenOutman — APPROVED** [review]
LGTM :)

**Comment by SevenOutman** [comment]
@hassanzohdy There seem be some conflicts after I merged your #2415 , could you please resolve them?

**Comment by hassanzohdy** [comment]
@SevenOutman LMAO, i thought i saw it in the file, just updated it again xD

**Pull request merged**